specfile/types.py

```diff
@@ -4,11 +4,14 @@
 import abc
 
 try:
-    from typing import SupportsIndex
+    from typing import SupportsIndex, TypedDict
```

Comment:
This shouldn't be here, although it's true that `TypedDict` doesn't exist in Python 3.6 either. Still, I wouldn't couple these two typed together and rather use a separate `try`-`except` for `TypedDict`.